### PR TITLE
Add tests for OMG.Block and extract Merkle-related functions

### PR DESCRIPTION
**Pull request by T-Dnzt**
## Overview

This PR adds tests for all public functions in `OMG.Block`. Functions in this module that were dealing with the `MerkleTree` library have also been extracted to a separate module, for easier testing and separation of concerns. 

I didn't remove any of the existing tests.

A few notes:

- The `to_db_value` and `from_db_value` don't do anything other than validating the type of the inputs and switching from map to struct and vice versa. I'm wondering if this kind of functions wouldn't make more sense in the `DB` sub-app.
- The `to_api_format` also feels a bit out of place to me, and since I'm just getting into the codebase, I'm not sure if it would make sense to have some formatters/serializers to put that logic into. 
- The functions in `Merkle` (moved from `Block`) have typespecs defining that that the passed list shouldn't be empty, but there isn't any actual check at the code level. Is that something we want to add?
- On the errors side, I've noticed there isn't any fallback functions for some of the functions checking the types of the given inputs, which results in `FunctionClauseError`. I wrote tests checking that because it seems to be the current expected behavior, but in some cases, we might want to add some fallbacks with proper errors - this would avoid `500` being returned to clients (we could return a `200` with an `internal_server_error` or something and properly log it).

## Changes

- Add a bunch of tests for `OMG.Block`
- Extract the functions dealing with the `MerkleTree` into a new module
- Add some tests for `OMG.Merkle`

## Testing

Simply run the test suite.

### CODE REVIEW

**Comment by InoMurko** [comment]
Goes well with https://github.com/omisego/elixir-omg/issues/901

File: apps/omg/test/omg/block_test.exs [hunk 1]
```diff
@@ -25,37 +25,227 @@ defmodule OMG.BlockTest do
...
+
+      assert %Block{transactions: [tx_1, tx_2, tx_3]}
+             |> Block.inclusion_proof(2)
+             |> Base.encode16(case: :lower) ==
+               "290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563c62f5e8b8c8303e7845c0e8ea800fc1bae9ccb7782b6f3c870c2d48fdddf5bb4890740a8eb06ce9be422cb8da5cdafc2b58c0a5e24036c578de2a433c828ff7d3b8ec09e026fdc305365dfc94e189a81b38c7597b3d941c279f042e8206e0bd8ecd50eee38e386bd62be9bedb990706951b65fe053bd9d8a521af753d139e2dadefff6d330bb5403f63b14f33b578274160de3a50df4efecf0e0db73bcdd3da5617bdd11f7c0a11f49db22f629387a12da7596f9d1704d7465177c63d88ec7d7292c23a9aa1d8bea7e2435e555a4a60e379a5a35f3f452bae60121073fb6eeade1cea92ed99acdcb045a6726b2f87107e8a61620a232cf4d7d5b5766b3952e107ad66c0a68c72cb89e4fb4303841966e4062a76ab97451e3b9fb526a5ceb7f82e026cc5a4aed3c22a58cbd3d2ac754c9352c5436f638042dca99034e836365163d04cffd8b46a874edf5cfae63077de85f849a660426697b06a829c70dd1409cad676aa337a485e4728a0b240d92b3ef7b3c372d06d189322bfd5f61f1e7203ea2fca4a49658f9fab7aa63289c91b7c7b6c832a6d0e69334ff5b0a3483d09dab4ebfd9cd7bca2505f7bef59cc1c12ecc708fff26ae4af19abe852afe9e20c8622def10d13dd169f550f578bda343d9717a138562e0093b380a1120789d53cf10"
```
[the remark this answers is not in the document]
Comment (T-Dnzt) [reply apps/omg/test/omg/block_test.exs]: Oh okay, wasn't sure which way to go, I'll fix it :)

File: apps/omg/test/omg/block_test.exs [hunk 2]
```diff
@@ -25,37 +25,227 @@ defmodule OMG.BlockTest do
...
+    } do
+      tx_1 = TestHelper.create_encoded([{1, 0, 0, alice}], eth(), [{alice, 7}])
+      tx_2 = TestHelper.create_encoded([{1, 1, 0, alice}], eth(), [{alice, 2}])
+
+      encoded_proof =
+        %Block{transactions: [tx_1, tx_2]}
+        |> Block.inclusion_proof(1)
+        |> Base.encode16(case: :lower)
+
+      assert "491b9e07c4997c976c38a91f" <> _ = encoded_proof
```
[the remark this answers is not in the document]
Comment (T-Dnzt) [reply apps/omg/test/omg/block_test.exs]: Yup, that looks nicer. Will update!

**Comment by T-Dnzt** [comment]
> looks like overzealous speccing, I can't recall a reason for nonempty(list). Especially for hash, not sure what create_tx_proof would yield in this case.

@pdobacz It actually works fine (as you can see [here](https://github.com/omisego/elixir-omg/pull/985/files#diff-7350974c5ddecbaf1d26cf05cf27b98aR204)), which is why I was curious what was the expected behavior. I'll update the typespecs.

**Comment by pdobacz** [comment]
> > looks like overzealous speccing, I can't recall a reason for nonempty(list). Especially for hash, not sure what create_tx_proof would yield in this case.
> 
> @pdobacz It actually works fine (as you can see [here](https://github.com/omisego/elixir-omg/pull/985/files#diff-7350974c5ddecbaf1d26cf05cf27b98aR204)), which is why I was curious what was the expected behavior. I'll update the typespecs.

oh, lol, right, so it allows one to prove inclusion of a default leaf :joy:. But then that's good, can remove typespecs I guess :).

**Pull request merged**